ocaml/xapi/xapi_cluster_helpers.ml

```diff
@@ -111,57 +111,54 @@ let cluster_health_enabled ~__context =
 
 let maybe_generate_alert ~__context ~num_hosts ~missing_hosts ~new_hosts ~quorum
     =
-  List.iter
-    (fun cluster_host ->
-      let host = Db.Cluster_host.get_host ~__context ~self:cluster_host in
-      let host_uuid = Db.Host.get_uuid ~__context ~self:host in
-      let host_name = Db.Host.get_name_label ~__context ~self:host in
-      let body =
-        Printf.sprintf
-          "Host %s has left the cluster, there are now %d host(s) in cluster \
-           and %d host(s) is required to form a quorum"
-          host_name num_hosts quorum
-      in
-      let name, priority = Api_messages.cluster_host_leaving in
-      Helpers.call_api_functions ~__context (fun rpc session_id ->
-          ignore
-          @@ Client.Client.Message.create ~rpc ~session_id ~name ~priority
-               ~cls:`Host ~obj_uuid:host_uuid ~body
-      )
+  let generate_alert join cluster_host =
+    let host = Db.Cluster_host.get_host ~__context ~self:cluster_host in
+    let host_uuid = Db.Host.get_uuid ~__context ~self:host in
+    let host_name = Db.Host.get_name_label ~__context ~self:host in
+    let body, name, priority =
+      match join with
+      | true ->
+        let body =
+          Printf.sprintf
+            "Host %s has joined the cluster, there are now %d host(s) in \
+             cluster and %d host(s) is required to form a quorum"
+            host_name num_hosts quorum
+        in
+        let name, priority = Api_messages.cluster_host_leaving in
+        (body, name, priority)
+      | false ->
+        let body =
+          Printf.sprintf
+            "Host %s has joined the cluster, there are now %d host(s) in \
+             cluster and %d host(s) is required to form a quorum"
+            host_name num_hosts quorum
+        in
+        let name, priority = Api_messages.cluster_host_joining in
+        (body, name, priority)
```

Comment:
We could make the code a little simpler:
```ocaml
    let body =
      Printf.sprintf
        "Host %s has joined the cluster, there are now %d host(s) in \
         cluster and %d host(s) is required to form a quorum"
        host_name num_hosts quorum in
    let name, priority =
      match join with
      | true ->
        Api_messages.cluster_host_leaving
      | false ->
        Api_messages.cluster_host_joining
```

Reply:
And I have a confused here:
If `join` is true, why dose not call `Api_messages.cluster_host_joining`?
And if `join` is false, we should call `Api_messages.cluster_host_leaving` in first impression by reading code.

Reply:
No you are absolutely right, many typos here, fixed them now.